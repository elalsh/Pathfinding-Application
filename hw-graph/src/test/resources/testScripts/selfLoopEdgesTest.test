# Test edges with 2 + self loop edges but with different labels


# Create a graph
CreateGraph graph1

# Add nodes A
AddNode graph1 A

# add an edge from A -> A (e1), A -> A (e2) and A -> A (e3)
AddEdge graph1 A A e1
AddEdge graph1 A A e2
AddEdge graph1 A A e3

# Print the nodes in the graph and the outgoing edges for src nodes to dest nodes
ListNodes graph1
ListChildren graph1 A

# Add a duplicate edge from A to A with label
AddEdge graph1 A A e1

# Print the nodes in the graph and the outgoing edges for src nodes to dest nodes
#Duplicate edges shouldn't modify the graph
ListNodes graph1
ListChildren graph1 A

