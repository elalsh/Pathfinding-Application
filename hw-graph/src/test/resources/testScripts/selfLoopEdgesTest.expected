# Test edges with 2 + self loop edges but with different labels


# Create a graph
created graph graph1

# Add nodes A
added node A to graph1

# add an edge from A -> A (e1), A -> A (e2) and A -> A (e3)
added edge e1 from A to A in graph1
added edge e2 from A to A in graph1
added edge e3 from A to A in graph1

# Print the nodes in the graph and the outgoing edges for src nodes to dest nodes
graph1 contains: A
the children of A in graph1 are: A(e1) A(e2) A(e3)

# Add a duplicate edge from A to A with label
added edge e1 from A to A in graph1

# Print the nodes in the graph and the outgoing edges for src nodes to dest nodes
#Duplicate edges shouldn't modify the graph
graph1 contains: A
the children of A in graph1 are: A(e1) A(e2) A(e3)
